### PR TITLE
fix: check if cursor is closed

### LINKED ISSUE #125: disable cursors methods after close()
At the moment it's possible to fetch from a closed cursor, but it shouldn't be allowed. From dbapi `close()` desciption:

> The cursor will be unusable from this point forward; an Error (or subclass) exception will be raised if any operation is attempted with the cursor.

See https://github.com/sqlalchemy/sqlalchemy/issues/6842#issuecomment-950125814

### PR DESCRIPTION
**Pull request by sureshdsk**
fixes #125 

@dvarrazzo not sure if this is the right way to fix. If this is not, your kind advice required. Thanks.

### CODE REVIEW

**Comment by sureshdsk** [comment]
Sure! I'm already working on tests. Will complete over weekend.

**Comment by dvarrazzo** [comment]
Tests seem good. You should rebase your branch: you can use the commands:
```
git fetch origin
git rebase origin/master
```
so you don't get spurious commits in this branch.

**Comment by sureshdsk** [comment]
@dvarrazzo thanks. rebased once. will do it again to see if everything is good

**Comment by sureshdsk** [comment]
@dvarrazzo also updated news. please let me know if anything else is required.

File: docs/news.rst [hunk 1]
```diff
@@ -9,6 +9,11 @@ Release notes
 
 Current release
 ---------------
+Psycopg 3.0.3
+^^^^^^^^^^^^^
+
+- Fix disable cursors methods after close() (:ticket:`#125`).
```
Comment (dvarrazzo) [inline docs/news.rst:15]: This will be released in 3.0.2

**Review by dvarrazzo — CHANGES_REQUESTED** [review]
The server-side cursors are affected by the same problem, but they fail differently:

```python
>>> import psycopg
>>> 
>>> cnn = psycopg.connect()
>>> cur = cnn.cursor("foo")
>>> cur.execute("select generate_series(1,10)")
<psycopg.ServerCursor 'foo' [COMMAND_OK] [INTRANS] (database=piro) at 0x7f6162553280>
>>> cur.fetchone()
(1,)
>>> cur.close()
>>> cur.fetchone()
Traceback (most recent call last):
[...]
psycopg.errors.InvalidCursorName: cursor "foo" does not exist
```

Could you please make them fail with `InterfaceError` too?

**Comment by sureshdsk** [comment]
@dvarrazzo fixed and added tests for server cursor and server async cursor too.

**Review by dvarrazzo — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -11,3 +11,4 @@ __pycache__/
 *.html
 /psycopg_binary/
 .vscode
+.venv
```
Comment (dvarrazzo) [inline .gitignore:14]: This file doesn't end in a newline. But no problem: I will make the fix in master and rebase this on, fixing the conflict.

**Comment by dvarrazzo** [comment]
Thank you very much! Will merge as soon as the tests pass.

File: .gitignore (hunk 2 above)
Comment (sureshdsk) [reply .gitignore:14]: cool

**Comment by sureshdsk** [comment]
@dvarrazzo Thank you for patiently guiding me 🙏🏽 Looking forward to contribute more! 🚀

**Pull request merged**